rust/definitions/src/navigation.rs

```diff
@@ -0,0 +1,665 @@
+pub struct MRecoverSeedName {
+    pub keyboard: bool,
+    pub seed_name: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MRecoverSeedPhrase {
+    pub keyboard: bool,
+    pub seed_name: String,
+    pub user_input: String,
+    pub guess_set: Vec<String>,
+    pub draft: Vec<String>,
+    pub ready_seed: Option<String>,
+}
+
+#[derive(Clone, Default, Debug, PartialEq)]
+pub struct DerivationCheck {
+    pub button_good: bool,
+    pub where_to: Option<DerivationDestination>,
+    pub collision: Option<Address>,
+    pub error: Option<String>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Address {
+    pub base58: String,
+    pub path: String,
+    pub has_pwd: bool,
+    pub identicon: Vec<u8>,
+    pub seed_name: String,
+    pub multiselect: Option<bool>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum DerivationDestination {
+    Pwd,
+    Pin,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MDeriveKey {
+    pub seed_name: String,
+    pub network_title: String,
+    pub network_logo: String,
+    pub network_specs_key: String,
+    pub suggested_derivation: String,
+    pub keyboard: bool,
+    pub derivation_check: Option<DerivationCheck>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct MVerifierDetails {
+    pub public_key: String,
+    pub identicon: Vec<u8>,
+    pub encryption: String,
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct MVerifier {
+    pub ttype: String,
+    pub details: MVerifierDetails,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MMetadataRecord {
+    pub specname: String,
+    pub specs_version: String,
+    pub meta_hash: String,
+    pub meta_id_pic: Vec<u8>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MNetworkDetails {
+    pub base58prefix: u16,
+    pub color: String,
+    pub decimals: u8,
+    pub encryption: Encryption,
+    pub genesis_hash: String,
+    pub logo: String,
+    pub name: String,
+    pub order: String,
+    pub path_id: String,
+    pub secondary_color: String,
+    pub title: String,
+    pub unit: String,
+    pub current_verifier: MVerifier,
+    pub meta: Vec<MMetadataRecord>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MRawKey {
+    pub seed_name: String,
+    pub address_key: String,
+    pub public_key: String,
+    pub identicon: Vec<u8>,
+    pub has_pwd: bool,
+    pub path: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MSignSufficientCrypto {
+    pub identities: Vec<MRawKey>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MKeyDetailsMulti {
+    pub key_details: MKeyDetails,
+    pub current_number: String,
+    pub out_of: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MMNetwork {
+    pub key: String,
+    pub title: String,
+    pub logo: String,
+    pub order: u8,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MManageNetworks {
+    pub networks: Vec<MMNetwork>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MSCContent {
```

Comment:
This was supposed to tell user what is being signed. Something like `Types` or `Network specs for westend` or `matedata 9090 for westend`. We could do an enum with empty variant for "types", "network" with its name and "metadata" with name and version.